apps/astarte_appengine_api/lib/astarte_appengine_api/auth/auth.ex

```diff
@@ -19,11 +19,15 @@
 defmodule Astarte.AppEngine.API.Auth do
   alias Astarte.AppEngine.API.Queries
   alias Astarte.DataAccess.Database
+  alias Astarte.AppEngine.API.Config
+  alias Astarte.Core.CQLUtils
 
   require Logger
 
   def fetch_public_key(realm) do
-    with {:ok, client} <- Database.connect(realm: realm),
+    keyspace_name = CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
```

Comment:
this is a valid point, however, starting with the fact that Config.astarte_instance_id! must not crash (and so, return always a valid value) because is validated at the service start, if for any circumstances it crashes it's a unexpected behavior (thus the !) and it must crash the entire application (and leave a meaningful stacktrace).

Expecting a tuple is not valid because we don't expect this function to ever return error and bubbling up. 

On the other hand, it's not fetch_public_key or it's logic that crashes, but a function inside it , otherwise every function (or most of them) shall end with ! and/or catch an eventual exception to bubble up